lbrynet/conf.py

```diff
@@ -64,86 +55,76 @@
 elif 'darwin' in sys.platform:
     platform = DARWIN
-    dirs = _get_old_directories(DARWIN)
 elif 'win' in sys.platform:
     platform = WINDOWS
-    if os.path.isdir(_get_old_directories(WINDOWS)['data']) or \
-       os.path.isdir(_get_old_directories(WINDOWS)['lbryum']):
-        dirs = _get_old_directories(WINDOWS)
-    else:
-        dirs = _get_new_directories(WINDOWS)
 else:
     platform = LINUX
-    if os.path.isdir(_get_old_directories(LINUX)['data']) or \
-       os.path.isdir(_get_old_directories(LINUX)['lbryum']):
-        dirs = _get_old_directories(LINUX)
-    else:
-        dirs = _get_new_directories(LINUX)
-
-default_data_dir = dirs['data']
-default_lbryum_dir = dirs['lbryum']
-default_download_dir = dirs['download']
 
 ICON_PATH = 'icons' if platform is WINDOWS else 'app.icns'
 
 
+def get_windows_directories() -> typing.Tuple[str, str, str]:
+    from lbrynet.winpaths import get_path, FOLDERID, UserHandle
+
+    download_dir = get_path(FOLDERID.Downloads, UserHandle.current)
+
+    # old
+    appdata = get_path(FOLDERID.RoamingAppData, UserHandle.current)
+    data_dir = os.path.join(appdata, 'lbrynet')
+    lbryum_dir = os.path.join(appdata, 'lbryum')
+    if os.path.isdir(data_dir) or os.path.isdir(lbryum_dir):
+        return data_dir, lbryum_dir, download_dir
+
+    # new
+    data_dir = user_data_dir('lbrynet', 'lbry')
+    lbryum_dir = user_data_dir('lbryum', 'lbry')
+    download_dir = get_path(FOLDERID.Downloads, UserHandle.current)
+    return data_dir, lbryum_dir, download_dir
+
+
+def get_darwin_directories() -> typing.Tuple[str, str, str]:
+    data_dir = user_data_dir('LBRY')
+    lbryum_dir = os.path.expanduser('~/.lbryum')
+    download_dir = os.path.expanduser('~/Downloads')
+    return data_dir, lbryum_dir, download_dir
+
+
+def get_linux_directories() -> typing.Tuple[str, str, str]:
+    download_dir = None
+    try:
+        with open(os.path.join(user_config_dir(), 'user-dirs.dirs'), 'r') as xdg:
+            down_dir = re.search(r'XDG_DOWNLOAD_DIR=(.+)', xdg.read()).group(1)
+            down_dir = re.sub('\$HOME', os.getenv('HOME'), down_dir)
+            download_dir = re.sub('\"', '', down_dir)
+    except EnvironmentError:
+        download_dir = os.getenv('XDG_DOWNLOAD_DIR')
+
+    if not download_dir:
+        download_dir = os.path.expanduser('~/Downloads')
+
+    # old
+    data_dir = os.path.expanduser('~/.lbrynet')
+    lbryum_dir = os.path.expanduser('~/.lbryum')
+    if os.path.isdir(data_dir) or os.path.isdir(lbryum_dir):
+        return data_dir, lbryum_dir, download_dir
+
+    # new
+    return user_data_dir('lbry/lbrynet'), user_data_dir('lbry/lbryum'), download_dir
+
+
+def get_android_directories(get_android_internal_storage: typing.Callable[[], str]) -> typing.Tuple[str, str, str]:
+    storage_dir = get_android_internal_storage()
+    data_dir = f'{storage_dir}/lbrynet'
+    lbryum_dir = f'{storage_dir}/lbryum'
+    download_dir = f'{storage_dir}/Download'
```

Comment:
Needs an additional parameter in order to get the private app directory where the `lbrynet` and `lbryum` folders will be created.